### PR TITLE
agentsvg: Functionality to generate all color and pose combinations

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: contrib/agentsvg/agentsvg.py [hunk 1]
```diff
@@ -55,32 +44,94 @@ def rect(x, y, w, h, color=cfengine_blue):
...
-<svg width="{}" height="{}" version="1.1" xmlns="http://www.w3.org/2000/svg">
-{}
-</svg>
-"""
+def line(x, y, dx, dy, n, radius, color):
+    """Returns a list of circles forming a line of length n"""
+    circles = []
+    if dx != 0 and dy != 0:
+        dx = dx * 0.70710678118
+        dy = dy * 0.70710678118
```
Comment (vpodzime) [inline contrib/agentsvg/agentsvg.py]: Please name the magic number

File: contrib/agentsvg/agentsvg.py [hunk 2]
```diff
@@ -55,32 +44,94 @@ def rect(x, y, w, h, color=cfengine_blue):
...
+def get_args():
+    ap = argparse.ArgumentParser(
+        description="Beautiful CFEngine agent logos",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    ap.add_argument("--radius", "-r", help="Circle raidus", type=int, default=16)
+    ap.add_argument("--body", help="Body color", type=str, default=cfengine_blue)
+    ap.add_argument("--head", help="Head color", type=str, default=cfengine_orange)
+    ap.add_argument("--arms", help="Arms pose", type=str, default="down")
+    ap.add_argument("--legs", help="Legs pose", type=str, default="straight")
```
Comment (vpodzime) [inline contrib/agentsvg/agentsvg.py]: These arguments could use `choices=[...]`.

File: contrib/agentsvg/agentsvg.py [hunk 3]
```diff
@@ -55,32 +44,94 @@ def rect(x, y, w, h, color=cfengine_blue):
...
+        content += line(6, 7, 1, 1, 4, radius, body)  # Right leg
+
+    content.append(Circle(5, 7, radius, body))  # Groin
+
+    # Offsets needed to center this logo in canvas:
+    offset_x = radius + diameter * 1
+    offset_y = radius
+
+    # Apply offsets to all circles:
+    for circle in content:
```
Comment (vpodzime) [inline contrib/agentsvg/agentsvg.py]: This applies to all shapes not just circles, right?

File: contrib/agentsvg/agentsvg.py [hunk 4]
```diff
@@ -55,32 +44,94 @@ def rect(x, y, w, h, color=cfengine_blue):
...
+
+    # The SVG container:
+    container = """
+    <svg width="{}" height="{}" version="1.1" xmlns="http://www.w3.org/2000/svg">
+    {}
+    </svg>
+    """
+
+    # Render SVG, print to stdout:
+    content = "\n".join([str(x) for x in content])
```
Comment (vpodzime) [inline contrib/agentsvg/agentsvg.py]: `[ ]` can be dropped here.

File: contrib/agentsvg/all_poses.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            arms = pose["arms"]
+            legs = pose["legs"]
+            pose_opts = f"--arms {arms} --legs {legs}"
+            command = f"python3 agentsvg.py {color_opts} {pose_opts} {radius_opt}"
+            base_name = f"agent_{head}_{body}_{arms}_{legs}_{dimensions}"
+            generate = f"{command} > ./{dimensions}/{base_name}.svg"
+            convert = f"convert ./{dimensions}/{base_name}.svg ./{dimensions}/{base_name}.png"
+            command = f"{generate} && {convert}"
+            print(command)
+            os.system(command)
```
Comment (vpodzime) [inline contrib/agentsvg/all_poses.py]: Maybe this could check the exit code?

File: contrib/agentsvg/all_poses.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+poses = [normal, out, angled, up, up_out]
+
+for size in sizes:
+    diameter = size * 2
+    width = diameter * 14
+    height = diameter * 12
+    dimensions = f"{width}x{height}"
+
+    radius_opt = f"--radius {size}"
+    os.system(f"mkdir -p ./{dimensions}")
```
Comment (vpodzime) [inline contrib/agentsvg/all_poses.py]: `os.makedirs(exist_ok=True)`.

File: contrib/agentsvg/agentsvg.py (hunk 3 above)
Comment (olehermanse) [reply contrib/agentsvg/agentsvg.py]: There are only circles. `rect()` generates a list of circles in a rectangular arrangement, `line()` generates a list of circles in a line.

File: contrib/agentsvg/agentsvg.py (hunk 2 above)
Comment (olehermanse) [reply contrib/agentsvg/agentsvg.py]: Thanks, added.

File: contrib/agentsvg/agentsvg.py (hunk 1 above)
Comment (olehermanse) [reply contrib/agentsvg/agentsvg.py]: Thanks, fixed.

File: contrib/agentsvg/agentsvg.py (hunk 4 above)
Comment (olehermanse) [reply contrib/agentsvg/agentsvg.py]: Fixed.

File: contrib/agentsvg/all_poses.py (hunk 5 above)
Comment (olehermanse) [reply contrib/agentsvg/all_poses.py]: Added.

File: contrib/agentsvg/all_poses.py (hunk 6 above)
Comment (olehermanse) [reply contrib/agentsvg/all_poses.py]: Thanks, fixed.

**Comment by olehermanse** [comment]
@vpodzime Thanks for your review, all comments addressed.

**Pull request merged**